include/wx/geometry.h

```diff
@@ -347,15 +360,39 @@ public :
         return wxPoint2DDouble( pt.m_x / n , pt.m_y / n );
     }
 
-    wxDouble m_x;
-    wxDouble m_y;
-};
+#if WXWIN_COMPATIBILITY_3_2
+    wxDEPRECATED_MSG("Multiplying points doesn't make sense")
+    inline wxPoint2DDouble& operator*=(const wxPoint2DDouble& pt)
+    {
+        m_x *= pt.m_x;
+        m_y *= pt.m_y;
+        return *this;
+    }
 
-inline wxPoint2DDouble::wxPoint2DDouble()
-{
-    m_x = 0.0;
-    m_y = 0.0;
-}
+    wxDEPRECATED_MSG("Multiplying points doesn't make sense")
```

Comment:
Why doesn't it make sense?
It makes sense when working with non-uniform scale factors, for example.

Reply:
Because methods like `GetVectorLength` `SetVectorLength`, `Normalize`, `GetDotProduct`, `GetCrossProduct` exist, looks like it was designed to store vector data as well.

Reply:
I mean, it _is_ a vector. But you don't multiply vectors like this.

Reply:
> Non uniform scale factor is not a point. Mathematically, points are elements of a vector space and there is no multiplication operation in it.

Absolutely agreed, you asked where this might have come from ;-) could be it was something from Taligent, I agree it is a misuse of a point struct to be a two-dimensional scalar factor

Reply:
`wxPoint2DDouble::operator*=(wxDouble)` is good enough for my purposes, but multiplying vectors like this look totally fine to me.

Reply:
Sorry, but this isn't a matter of opinion, I'm speaking in mathematical sense and you just can't define multiplication like this in a vector space... To convince you of this, consider that `(1,0)*(0,1) == (0,0)` with this "multiplication" definition, so you can have `x*y = 0` with `x ≠ 0` and `y ≠ 0` (and, relatedly, neither of these elements has an inverse).